perl/shop/t/Basket.t

```diff
@@ -1,319 +1,364 @@
 use strict;
-use Test::More tests => 122;
-use Data::Dumper;
+use Test::More;
 use WebServiceClient;
-use WebServiceConfiguration qw( WEBSERVICE_URL WEBSERVICE_USER WEBSERVICE_SHOP_PATH);
-use WebServiceTools qw( TAttributes hAttributes cmpDateTime );
+use WebServiceConfiguration qw( WEBSERVICE_URL WEBSERVICE_USER );
 
 # Create a SOAP::Lite client object
-my $BasketService = WebServiceClient
-    ->uri( 'urn://epages.de/WebService/BasketService/2013/11' )
-    ->proxy( WEBSERVICE_URL )
-    ->userinfo( WEBSERVICE_USER )
-    ->xmlschema('2001');
+my $BasketService =
+  WebServiceClient->uri('urn://epages.de/WebService/BasketService/2013/11')->proxy(WEBSERVICE_URL)
+  ->userinfo(WEBSERVICE_USER)->xmlschema('2001');
 
 # use product service to get GUIDs by given object paths of demo products
-my $ProductService = WebServiceClient
-    ->uri( 'urn://epages.de/WebService/ProductService/2013/01' )
-    ->proxy( WEBSERVICE_URL )
-    ->userinfo( WEBSERVICE_USER )
-    ->xmlschema('2001');
+my $ProductService =
+  WebServiceClient->uri('urn://epages.de/WebService/ProductService/2013/01')->proxy(WEBSERVICE_URL)
+  ->userinfo(WEBSERVICE_USER)->xmlschema('2001');
+
+my $Product_alias_1 = "productalias1";
+my $Product_alias_2 = "productalias2";
+
+# create test products
+my $Product_in_1 = {
+    'Alias'         => $Product_alias_1,
+    'StockLevel'    => 200,
+    'ProductPrices' => [ { 'CurrencyID' => 'EUR', 'Price' => 123, 'TaxModel' => 'gross', }, ],
+    'IsAvailable'   => SOAP::Data->type('boolean')->value(1)
+};
+my $Product_in_2 = {
+    'Alias'         => $Product_alias_2,
+    'StockLevel'    => 300,
+    'ProductPrices' => [ { 'CurrencyID' => 'EUR', 'Price' => 1234, 'TaxModel' => 'gross', }, ],
+    'IsAvailable'   => SOAP::Data->type('boolean')->value(1)
+};
 
-# array of product paths that involved in basket tests
-my @ProductPaths = map "Products/$_", qw(ho_1112105010 eg_1000111010 de_3201212002);
-my $prodInfoResult = $ProductService->getInfo(\@ProductPaths,['GUID'])->result;
-    ok( !$prodInfoResult->[0]->{'Error'}, "getInfo product GUIDs: no error" );
-# store GUID for easy acces via $GUID{$Alias}
-my %GUID = map { $_->{Alias} => $_->{Attributes}->[0]->{Value} } @$prodInfoResult;
+sub main {
+    init();
+    test();
+    done_testing();
+}
 
+sub init() {
+    _cleanup();
+}
 
-sub testExistsByPath {
-    my ( $Path, $exists ) = @_;
+sub test {
+    testCreateBasket();
+    testExistsBasket();
+    testDeleteBasket();
+    testaddProductLineItem();
+    testGetInfoReference();
+    testUpdateLineItem();
+    testDeleteLineItem();
+    testUpdateBasket();
+}
 
-    my $ahResults = $BasketService->exists( [$Path] )->result;
-    is( scalar @$ahResults, 1, 'exists: result count' );
+sub testCreateBasket {
```

Comment:
Test seems to solely test the response of the web service call. It doesn't check if a basket was successfully created.

I think, since this is an integration test, a call to `$BasketService->getInfo` should be added. The result of `getInfo` could be compared to `$hExpBasket` to ensure a successful roundtrip of the data.